src/components/Banner/Banner.tsx

```diff
@@ -55,202 +59,210 @@ export interface BannerProps {
   stopAnnouncements?: boolean;
 }
 
-export class Banner extends PureComponent<BannerProps, State> {
-  static contextType = FeaturesContext;
-  context!: React.ContextType<typeof FeaturesContext>;
-
-  state: State = {
-    showFocus: false,
-  };
+interface BannerAttributes {
+  iconColor: IconProps['color'];
+  defaultIcon: IconProps['source'];
+  ariaRoleType?: 'alert';
+}
 
-  private wrapper = createRef<HTMLDivElement>();
+function useBannerAttributes(status: BannerProps['status']): BannerAttributes {
```

Comment:
Thoughts on passing a `newDesignLanguage` boolean arg into this and then it can be a regular function instead of a hook?

Reply:
⚠️  Opinions ahead 😄 

IMO passing hook values as arguments/props is kind of an anti-pattern since hooks are great “dependency injectors”. Reducing the number of attributes you need to pass around leads to better APIs and less “cognitive load” (obviously the complexity of that hook isn’t such a big issue here, but I still tend to always do it).
I guess it also has the benefit of allowing you to re-use that hook in other places without having to care about its internal implementation, and allows for simpler composition with other hooks.

One could argue this leads to more calls then needed but `useFeatures` only reads from context, which is cheap. Even if it somehow computed something expensive, it should be handled inside it (like with a `useMemo`), it should not be the consumer’s responsibility to think about performance.

---

I don’t feel strongly about it so happy to make the change, though!

Reply:
Cool, that all makes sense to me :)

> Reducing the number of attributes you need to pass around leads to better APIs and simpler usage.

It leads to simpler usage but IIMO it can potentially hide the complexity around what something depends upon. With "pure" functions dependencies are simple - the arguments of the functions; once you move away from that by using hooks for DI then IMO that relationship can sometimes get a bit harder to see and now folks have to understand the additional complexity of knowing "you also need to provide this context" which isn't apparent from skimming over an argument list - so there's a tradeoff between "easier if you've already got the context provider setup" and "harder to setup the first time as you need to work out how the context provider does its thing". That's not an issue here as there's already an expectation of "you need to wrap everything in an AppProvider" for polaris-react components, but might be something to consider of when building other APIs :)

This particular case is very low-key - its a single arg, and the context will always already be provided so there's nothing new to teach there - so I don't have strong opinions either way. I'm happy to leave it as it is :)

Reply:
> It leads to simpler usage but IIMO it can potentially hide the complexity around what something depends upon.

Right, you got me on that one. 😄 

I made the change!